lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -108,6 +110,14 @@ defmodule ExUnit.Assertions do
     vars = collect_vars_from_pattern(left)
     pins = collect_pins_from_pattern(left, Macro.Env.vars(__CALLER__))
 
+    left_pattern = escape_quoted(:pattern, left)
```

Comment:
We should have a `escape_pattern` function instead of wrapping this in `pattern` just to immediately remove it.